File: src/Microsoft.AspNetCore.SignalR.Client/HttpConnectionOptions.cs
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.

using System;
using System.Collections.Generic;
using System.Net;
using System.Net.Http;
using System.Net.WebSockets;
using System.Security.Cryptography.X509Certificates;
using Microsoft.AspNetCore.Http.Connections;

namespace Microsoft.AspNetCore.SignalR.Client
{
    public class HttpConnectionOptions
    {
        internal X509CertificateCollection _clientCertificates;
        internal IDictionary<string, string> _headers;
        internal CookieContainer _cookies;

        public Uri Url { get; set; }
        public HttpTransportType? Transport { get; set; }
        public Func<HttpMessageHandler, HttpMessageHandler> MessageHandlerFactory { get; set; }
        public bool? UseDefaultCredentials { get; set; }
        public ICredentials Credentials { get; set; }
        public IWebProxy Proxy { get; set; }
        public Func<string> AccessTokenFactory { get; set; }
        public Action<ClientWebSocketOptions> WebSocketOptions { get; set; }

        public X509CertificateCollection ClientCertificates
        {
            get
            {
                if (_clientCertificates == null)
                {
                    _clientCertificates = new X509CertificateCollection();
                }

                return _clientCertificates;
            }
        }

        public CookieContainer Cookies
        {
            get
            {
                if (_cookies == null)
                {
                    _cookies = new CookieContainer();
                }

                return _cookies;
            }
        }

        public IDictionary<string, string> Headers
        {
            get
            {
                if (_headers == null)
                {
                    _headers = new Dictionary<string, string>();
                }

                return _headers;
            }
        }
    }
}

File: clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HttpConnectionOptions.java
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.

package com.microsoft.aspnet.signalr;

public class HttpConnectionOptions {
    private String url;
    private Transport transport;
    private LogLevel loglevel;

    private Logger logger;
    private boolean skipNegotiate;

    public HttpConnectionOptions() {}

    public HttpConnectionOptions(String url, Transport transport, LogLevel logLevel, boolean skipNegotiate) {
        this.url = url;
        this.transport = transport;
        this.skipNegotiate = skipNegotiate;
        this.loglevel = logLevel;
    }

    public HttpConnectionOptions(String url, Transport transport, Logger logger, boolean skipNegotiate) {
        this.url = url;
        this.transport = transport;
        this.skipNegotiate = skipNegotiate;
        this.logger = logger;
    }
    public void setUrl(String url) {
        this.url = url;
    }

    public void setTransport(Transport transport) {
        this.transport = transport;
    }

    public void setLoglevel(LogLevel loglevel) {
        this.loglevel = loglevel;
    }

    public void setSkipNegotiate(boolean skipNegotiate) {
        this.skipNegotiate = skipNegotiate;
    }

    public String getUrl() {
        return url;
    }

    public Transport getTransport() {
        return transport;
    }

    public LogLevel getLoglevel() {
        return loglevel;
    }

    public boolean getSkipNegotiate() {
        return skipNegotiate;
    }

    public Logger getLogger() {
        return logger;
    }

    public void setLogger(Logger logger) {
        this.logger = logger;
    }
}
